build/scripts/cache_projects.sh

```diff
@@ -34,7 +34,11 @@ function cache_project() {
   local sparse_checkout_dir="$4"
 
   rm -fr "$TEMP_REPO"
-  git clone "$repo" -b "$branch" --depth 1 "$TEMP_REPO" -q
+  git clone "$repo" -b "$branch" --depth 1 "$TEMP_REPO" -q || (
```

Comment:
If I understand this correctly, we're doing the second part if the first `git clone` fails for any reason. IMO, it would be cleaner to check explicitly whether `$branch` is empty. The first clone may fail for any number of reasons.

Reply:
It may be misleading though in this case `$branch` is not empty. It contains the commit hash.
`git` can't clone with commit ID. This is the  reason I added the second part.

IMO, `$branch` is no always empty value. Since `$branch` is set as `master` before coming here if `$branch` was empty.

Reply:
I've opened https://github.com/eclipse/che-devfile-registry/pull/258 as alternate fix to really clone with the expected sha-1